chef/cookbooks/ses/providers/config.rb

```diff
@@ -0,0 +1,88 @@
+action :create do
+  create_configs(new_resource.name)
+end
+
+def supported_services
+  ["glance", "cinder", "nova"]
+end
+
+# TODO: use 'ceph' group for ownership of keyring files and add relevant service users
+# to this group instead of using per-service groups. This should cover case where multiple
+# services use the same user/keyring.
+
+# TODO: do we need some service restarting/reloading logic here? maybe some
+# (smart) detection if configs were changed? or is this handled by template somehow?
+
+def write_keyring_file(client_name, keyring_path, key, group_name)
+  Chef::Log.info("SES create #{client_name} keyring #{keyring_path}")
+
+  template keyring_path do
+    cookbook "ses"
+    source "client.keyring.erb"
+    owner "root"
+    group group_name
```

Comment:
The group name should just be ceph, and all of the service users should be added to the ceph group.
Folks can hand edit the yaml file and make all the services use a single client, so you'd end up with glance using the cinder key, so glance needs permissions to read the keyring file.  If they are all part of the same group, then the services can read those keyring files.

After running this the first time and all the services are on the same controller, each keyring ends up with a cinder group permission, which is why glance is failing to create an image.

```
root@d52-54-77-77-01-01:/etc/ceph # ls -al
total 36
drwxr-xr-x   2 root root    4096 Feb 27 14:04 .
drwxr-xr-x 121 root root   12288 Feb 27 14:12 ..
-rw-r-----   1 root cinder    17 Feb 27 14:04 ceph.client..keyring
-rw-r-----   1 root cinder    68 Feb 27 13:52 ceph.client.walt-cinder.keyring
-rw-r-----   1 root cinder    68 Feb 27 13:39 ceph.client.walt-glance.keyring
-rw-r--r--   1 root root     788 Feb 27 14:04 ceph.conf
-rw-r--r--   1 root root      92 Feb 12 17:41 rbdmap

Reply:
should be fixed now